src/components/Radio/Radio.test.tsx

```diff
@@ -0,0 +1,104 @@
+import { expect } from "chai";
+import * as Enzyme from "enzyme";
+import * as React from "react";
+import { stub } from "sinon";
+
+import Radio from "./Radio";
+
+describe("Radio Button", () => {
+  let container;
+  let changeHandler;
+  let clickHander;
+
+  before(() => {
+    clickHander = stub();
+    changeHandler = stub();
+    container = Enzyme.mount(
+      <Radio
+        id="inputID"
+        attributes={{ onClick: clickHander }}
+        onChange={changeHandler}
+      ></Radio>
+    );
+  });
+
+  it("Renders without crashing", () => {
+    expect(container.find("input").exists()).to.equal(true);
+  });
+
+  it("The radio element is an input with type='radio'", () => {
+    expect(container.find("input").prop("type")).to.equal("radio");
+  });
+
+  it("Allows user to pass in additional attributes", () => {
+    container.simulate("click");
+    expect(clickHander.callCount).to.equal(1);
+  });
+
+  it("Changing the value calls the onChange handler", () => {
+    container.find("input").simulate("change", { target: { value: "Hello" } });
+    expect(changeHandler.callCount).to.equal(1);
+  });
+
+  it("Renders with label", () => {
+    container = Enzyme.mount(
+      <>
+        <Radio
+          //add other props here
+          labelText="Hello"
+        ></Radio>
+      </>
+    );
+    expect(container.find("label").exists()).to.equal(true);
+  });
+
+  it("The 'checked' attribute can set properly", () => {
+    const onChange = stub();
+    const container = Enzyme.mount(
+      <Radio
+        id="inputID-attributes"
+        attributes={{
+          checked: true,
+          "aria-checked": true,
+          onChange,
+        }}
+      ></Radio>
+    );
+
+    const input = container.find("input");
+
+    expect(input.prop("checked")).to.equal(true);
+    expect(input.prop("aria-checked")).to.equal(true);
+  });
+
+  it("Passes the ref to the input element", () => {
+    const ref = React.createRef<HTMLInputElement>();
+    const container = Enzyme.mount(
+      <Radio id="inputID-attributes" ref={ref}></Radio>
+    );
+    expect(container.find("input").instance()).to.equal(ref.current);
+  });
+
+  it("Renders HTML attributes passed through the `attributes` prop", () => {
+    const onChangeSpy = stub();
+    const onBlurSpy = stub();
+    container = Enzyme.mount(
+      <Radio
+        id="inputID-attributes"
+        attributes={{
+          onChange: onChangeSpy,
+          onBlur: onBlurSpy,
+          maxLength: 10,
```

Comment:
This is a bit weird to pass because it doesn't do anything to a radio type input.

Reply:
Fixed!